app/models/request.rb

```diff
@@ -20,6 +20,16 @@ class Request < ApplicationRecord
 
   before_create :set_context
 
+  def switch_context
+    if context.present?
+      ManageIQ::API::Common::Request.with_request(context.transform_keys(&:to_sym)) do
+        ActsAsTenant.with_tenant(tenant) { yield }
+      end
+    else
+      ActsAsTenant.with_tenant(tenant) { yield }
```

Comment:
@bzwei We should never be without a context, that is an error case and we should raise an exception so that we can trap all these code paths early on.